### PR TITLE
Subpackage targets

### PR DESCRIPTION
**Pull request by hovind**
Trying to call `build_openroad()` from subpackages does not currently work. I've drafted a sketch to make it work when called from subpackages of `bazel-orfs`.

It still does not work when `build_openroad()` is called from a subpackage in an external project, with the following error message from `docker_shell.sh`, I believe it has to do with how `WORKSPACE_ROOT` is defined.

```
+ chmod -R +w /home/.../execroot/_main/execroot/_main/bazel-out/k8-fastbuild/bin
chmod: cannot access '/home/.../execroot/_main/execroot/_main/bazel-out/k8-fastbuild/bin
```

### CODE REVIEW

File: openroad.bzl [hunk 1]
```diff
@@ -592,7 +592,7 @@ def build_openroad(
     stage_args = init_stage_dict(all_stage_names, stage_args)
     stage_args["clock_period"] = SDC_FILE
     stage_args["synth_sdc"] = SDC_FILE
-    stage_args["synth"].append("VERILOG_FILES=" + " ".join(verilog_files))
+    stage_args["synth"].append("VERILOG_FILES=" + " ".join(map(lambda v: native.package_name() + "/" + v, verilog_files)))
```
Comment (hovind) [inline openroad.bzl:595]: I believe this is too simple if we want to pass in labels that are outputs of other rules.

**Comment by oharboe** [comment]
@lpawelcz @eszpotanski Merging for now, but this needs a better solution.

Thoughts?

**Comment by lpawelcz** [comment]
@hovind thanks for adding CI test case!

**Comment by lpawelcz** [comment]
> It still does not work when build_openroad() is called from a subpackage in an external project, with the following error message from docker_shell.sh, I believe it has to do with how WORKSPACE_ROOT is defined.

I was able to get this to work in external workspace (megaboom). Please take a look at: https://github.com/The-OpenROAD-Project/megaboom/pull/59

**Comment by hovind** [comment]
> > It still does not work when build_openroad() is called from a subpackage in an external project, with the following error message from docker_shell.sh, I believe it has to do with how WORKSPACE_ROOT is defined.
> 
> I was able to get this to work in external workspace (megaboom). Please take a look at: [The-OpenROAD-Project/megaboom#59](https://github.com/The-OpenROAD-Project/megaboom/pull/59)

Thanks! Not at my computer right now, but I will test first thing in the morning!

Did you try with a `local_path_override()` of `bazel-orfs`? That might have been my problem.

**Comment by lpawelcz** [comment]
Yes, I also tested setup with local `bazel-orfs` and it did work

**Comment by hovind** [comment]
> Yes, I also tested setup with local `bazel-orfs` and it did work

Thanks, pinning bazel to `7.0.0` instead of using `7.1.2` works!

**Pull request merged**